export/export_pipeline/compute_collision_representation.py

```diff
@@ -0,0 +1,427 @@
+    args: a tuple containing (path to mesh, hyperparameters, rerun flag).
+    Returns a bool indicating success of the operation.
+    """
+    mesh_path, params, rerun = args
+    mesh_dir, mesh_filename = osp.split(mesh_path)
+    vhacd_args = params['collision_representation']['vhacd']
+    coacd_args = params['collision_representation']['coacd']
+    acd_args   = params['collision_representation']['acd']
+
+    # Copy the obj file to the temporary directory.
+    decomposed = False
+    decompose_dir = osp.join(mesh_dir, params['common']['CONVEX_DECOMPOSITION_DIR'])
+    if rerun and osp.isdir(decompose_dir):
+        decomposed = True
+        os.remove(mesh_path)
+    else:
+        os.makedirs(decompose_dir, exist_ok=True)
+        mesh_path = shutil.move(mesh_path, decompose_dir)
+        obj_name = os.path.splitext(mesh_filename)[0]
+        
+        # load mesh and check that it is valid
+        mesh: trimesh.Trimesh = trimesh.load(mesh_path, force='mesh')
+        if mesh.is_empty:
+            volume = 0.0
+        else:
+            try:
+                volume = mesh.bounding_box_oriented.volume
+            except ValueError:
+                volume = 0.0
+        if volume < 1e-6:
+            acd_args['decompose_method'] = 'skip'
+            print(f'{obj_name} is too small, volume = {volume:.5f}, will skip')
+        elif volume < acd_args['min_volume']:
+            acd_args['decompose_method'] = 'none'
+            print(f'{obj_name} is small, volume = {volume:.5f}, will not decompose')
+
+        # save convex hull
+        hull_mesh: trimesh.Trimesh = trimesh.convex.convex_hull(mesh)
+        hull_filename = osp.join(mesh_dir, 'convex_hull.stl')
+        hull_mesh.export(hull_filename)
+        print(f'Convex hull {hull_filename} saved.')
+        
+        if acd_args['decompose_method'] == 'coacd':
+            # Call CoACD.
+            if coacd_args['auto_threshold']:
+                model = coacd_args['threshold_model']
+                threshold = max(0.02, volume*model[0] + model[1])
+            else:
+                threshold = coacd_args['threshold']
+            mesh = coacd.Mesh(mesh.vertices, mesh.faces)
+            result = coacd.run_coacd(
+                mesh,
+                threshold=threshold,
+                max_convex_hull=coacd_args['max_convex_hull'],
+                preprocess_mode=coacd_args['preprocess_mode'],
+                preprocess_resolution=coacd_args['preprocess_resolution'],
+                resolution=coacd_args['resolution'],
+                mcts_nodes=coacd_args['mcts_nodes'],
+                mcts_iterations=coacd_args['mcts_iterations'],
+                mcts_max_depth=coacd_args['mcts_max_depth'],
+                pca=coacd_args['pca'],
+                merge=coacd_args['merge'],
+                seed=coacd_args['seed'],
+                )
+
+            mesh_parts = []
+            for vs, fs in result:
```

Comment:
- These variable names are somewhat confusing because they're so terse.
- Convert `vs` and `fs` to numpy arrays so you can do this cleanup step with numpy fancy indexing, rather than resorting to slow ugly Python for-loops.
- Python List comprehensions are more elegant than raw for loops but will still be slow for large meshes.